### PR TITLE
Batch stories-visible information in the collections endpoint response

### PR DESCRIPTION
**Pull request by jonathonherbert**
This reduces the number of gets we make when we're asking for collections:

Previously -
- GET for n collectionIds returns collections
- (GET for collection story count returns stories visible) for each collection (possibly more, given the numbers below for the UK CODE front? Bit confused!)
- screen shot 2019-01-22 at 13 31 07
(NB: this includes a few visual assets to render the collections as well as the collection request)

Now -
- GET for n collectionIds returns collections + stories visible
- screen shot 2019-01-22 at 13 30 26

Tested on CODE. We'd really appreciate comments about the idiomatic-ness (idiomaticity?) and readability of the Scala!

### CODE REVIEW

File: app/controllers/FaciaToolController.scala [hunk 1]
```diff
@@ -55,25 +56,6 @@ class FaciaToolController(
     }
   }
```
Comment (jonathonherbert) [inline app/controllers/FaciaToolController.scala]: To make the endpoint usage across tool versions explicit, we've moved getCollections to the V2 controller.

File: app/controllers/StoriesVisibleController.scala [hunk 2]
```diff
@@ -11,57 +12,11 @@ case class StoriesVisibleRequest(
...
-  implicit val jsonFormat = Json.format[StoriesVisibleResponse]
-}
-
-case class StoriesVisibleResponse(
-  desktop: Option[Int],
-  mobile: Option[Int]
-)
-
-class StoriesVisibleController(val containers: Containers, val deps: BaseFaciaControllerComponents) extends BaseFaciaController(deps) {
+class StoriesVisibleController(val containerService: ContainerService, val deps: BaseFaciaControllerComponents) extends BaseFaciaController(deps) {
```
Comment (jonathonherbert) [inline app/controllers/StoriesVisibleController.scala]: The logic here now lives in a service, and the controller is nice and slim as a result.

File: client-v2/src/actions/Collections.ts [hunk 3]
```diff
@@ -48,13 +48,34 @@ function getCollections(collectionIds: string[]): ThunkResult<Promise<void>> {
...
       await Promise.all(
-        collections.map(async collection => {
+        collectionResponses.map(async (collectionResponse, index) => {
+          const collectionId = collectionIds[index];
           const collectionConfig = getCollectionConfig(
             getState(),
-            collection.id
+            collectionId
           );
+          if (!collectionResponse) {
```
Comment (jonathonherbert) [inline client-v2/src/actions/Collections.ts]: There are a few additional cases to manage here, as any result in an array of returned collections can be null.

File: test/services/CollectionServiceTest.scala [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package services
+
+import com.gu.facia.client.models.{CollectionConfigJson, CollectionJson, ConfigJson, FrontJson, Trail}
+import conf.ApplicationConfiguration
+import org.joda.time.DateTime
+import org.scalatest.{DoNotDiscover, FreeSpec, Matchers}
+import play.api.{Configuration}
+import slices.{Containers, FixedContainers, Story}
+
+@DoNotDiscover class CollectionServiceTest extends FreeSpec with Matchers {
```
Comment (akash1810) [inline test/services/CollectionServiceTest.scala]: why the `DoNotDiscover` annotation?

File: test/services/CollectionServiceTest.scala [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    val draft = Trail("newId", 0, Some(""), None) :: live
+    CollectionJson(live, Some(draft), None, new DateTime(0), "oldUpdatedBy", "oldUpdatedEmail", None, None, None)
+  }
+
+  private def configJson: ConfigJson = {
+    ConfigJson(Map[String, FrontJson](), Map[String, CollectionConfigJson]("id" -> collectionConfig))
+  }
+
+  private def collectionConfig: CollectionConfigJson = {
+    CollectionConfigJson(
```
Comment (akash1810) [inline test/services/CollectionServiceTest.scala]: do we do this often? that is, construct a largely empty `CollectionConfigJson`? If so, we could push its creation to the companion object to make things a bit DRYer.

File: test/services/CollectionServiceTest.scala (hunk 4 above)
Comment (jonathonherbert) [reply test/services/CollectionServiceTest.scala]: Because it was `ctrl-c`'ed from another test class! Whoops! We can remove it.

**Pull request merged**